src/pkg_config_gen.rs

```diff
@@ -307,6 +282,102 @@ mod test {
         );
         pkg.add_lib("-lbar").add_cflag("-DFOO");
 
-        println!("{:?}\n{}", pkg, pkg.render());
+        let expected = concat!(
+            "prefix=/usr/local\n",
+            "exec_prefix=${prefix}\n",
+            "libdir=${exec_prefix}/lib\n",
+            "includedir=${prefix}/include\n",
+            "\n",
+            "Name: foo\n",
+            "Description: \n",
+            "Version: 0.1\n",
+            "Libs: -L${libdir} -lfoo -lbar\n",
+            "Cflags: -I${includedir} -DFOO\n",
+            "Requires: somelib, someotherlib\n",
+            "Requires.private: someprivatelib >= 1.0\n",
+        );
+
+        assert_eq!(expected, pkg.render());
+    }
+
+    mod test_canonicalize {
+        use super::canonicalize;
+
+        #[test]
+        fn test_absolute_path() {
+            let path = "/home/user/docs";
+            let result = canonicalize(path);
+            assert_eq!(result, "/home/user/docs");
+        }
+
+        #[test]
+        fn test_relative_path() {
+            let path = "home/user/docs";
+            let result = canonicalize(path);
+            assert_eq!(result, "home/user/docs");
+        }
+
+        #[test]
+        fn test_current_directory() {
+            let path = "/home/user/./docs";
+            let result = canonicalize(path);
+            assert_eq!(result, "/home/user/docs");
+        }
+
+        #[test]
+        fn test_parent_directory() {
+            let path = "/home/user/../docs";
+            let result = canonicalize(path);
+            assert_eq!(result, "/home/user/../docs");
+        }
```

Comment:
is this expected? should that not output `/home/docs`?

Reply:
It should, if you want to add more logic is great :)

Reply:
I'll play around with it a bit

Reply:
got it to work. less pretty now but more functionality. I think this still gives weird results for e.g. `../` or any case where there are more parent directory components than there are normal components. should that panic maybe? or is that unlikely to come up?

Reply:
Its origin is to canonicalize \-paths to /-paths so do not dwell on it your original set + the clippy fix would be already great.